src/Application/Utility/EmailSender.cs

```diff
@@ -47,10 +54,24 @@ public async Task SendEmailConfirmationEmail(User user, string token)
 
     public async Task SendPasswordResetEmail(User user, string token)
     {
-        var resetLink = $"https://librumreader.com/resetPassword?email={user.Email}&token={token}";
-        
+		// go to librum site if not selfhosted
+		var resetLink = $"https://librumreader.com/resetPassword?email={user.Email}&token={token}";
+		// if self hosted change resetlink
+		if (_configuration["LIBRUM_SELFHOSTED"] == "true"){
+			var domain =_configuration["CleanUrl"];
+			var encodedToken=System.Web.HttpUtility.HtmlEncode(token);
+			resetLink = $"{domain}/user/resetPassword?email={user.Email}&token={encodedToken}";
```

Comment:
I don't understand what you are doing here. Could you give me an example?

Reply:
For example:
1. self-hosted server is listening to https  127.0.0.1:5001 , 
2. it's exposed to network as mybook.com (standart port 443)
3. _configuration["CleanUrl"]  = https://mybooks.com, 
4. user will get email with password  reset link starting with mybooks.com/user/resetPassword.....  (not   127.0.0.1:5001).  
---------------
PS. its possible that server isn't aware about  its public url . Server running on 127.0.0.1:5001 and exposed to web by nginx.
So , the _configuration["CleanUrl"] is used to build "password reset" link with "public url".

Reply:
I understand. Thanks!